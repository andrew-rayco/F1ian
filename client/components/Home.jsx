import React from 'react'

import * as api from '../api'

export default class Home extends React.Component {
  constructor() {
    super()
    this.state = {
      seasons: null
    }
  }

  componentWillMount() {
    api.getSeasons((seasons) => {
      this.setState({seasons})
    })
  }

  listSeasons(seasons) {
    return seasons.map((item) => {
      let year = item.year
      return <li key={year}><a href={`api-v1/season/${year}`}>{year}</a></li>
    })
  }

  render() {
    return (
        <div className="row">
          <div className="twelve columns home">
            <h3>Seasons</h3>
            <ul>
              {this.state.seasons ? this.listSeasons(this.state.seasons) : 'loading...'}
            </ul>

            <h3>Circuits</h3>
            <div className="circuits">
              <a href="#/circuits">See all the circuits</a>
            </div>
          </div>
        </div>
    )
  }

}
