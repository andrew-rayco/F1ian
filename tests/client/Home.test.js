import React from 'react'
import { shallow } from 'enzyme'

import Home from '../../client/components/Home'

const data = [
  {
    "season": "2010",
    "url": "https://en.wikipedia.org/wiki/2010_Formula_One_season"
  },
  {
    "season": "2011",
    "url": "https://en.wikipedia.org/wiki/2011_Formula_One_season"
  },
  {
    "season": "2012",
    "url": "https://en.wikipedia.org/wiki/2012_Formula_One_season"
  },
  {
    "season": "2013",
    "url": "https://en.wikipedia.org/wiki/2013_Formula_One_season"
  },
  {
    "season": "2014",
    "url": "https://en.wikipedia.org/wiki/2014_Formula_One_season"
  },
  {
    "season": "2015",
    "url": "https://en.wikipedia.org/wiki/2015_Formula_One_season"
  },
  {
    "season": "2016",
    "url": "https://en.wikipedia.org/wiki/2016_Formula_One_season"
  },
  {
    "season": "2017",
    "url": "https://en.wikipedia.org/wiki/2017_Formula_One_season"
  },
  {
    "season": "2018",
    "url": "https://en.wikipedia.org/wiki/2018_Formula_One_World_Championship"
  },
  {
    "season": "2019",
    "url": "https://en.wikipedia.org/wiki/2019_Formula_One_World_Championship"
  }
]

describe('Home component', () => {
    const wrapper = shallow(<Home />)

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('contains a seasons and circuits h3 heading', () => {
        expect(wrapper.find('h3').length).toEqual(2)
        expect(wrapper.find('[data-test="heading-seasons"]').text()).toEqual('Seasons')
        expect(wrapper.find('[data-test="heading-circuits"]').text()).toEqual('Circuits')
    })

    it('renders Loading component if no season data (yet)', () => {
        const component = wrapper.find('ul')
        expect(component.children().length).toEqual(1)
        expect(component.children().text()).toEqual('<Loading />')
    })

    it('renders seasons if given data and does not render Loading component', () => {
        wrapper.setState({ seasons: data })
        const component = wrapper.find('ul')
        expect(component.children().length).toBe(10)
        expect(component.contains('<Loading />')).toBeFalsy()
    })
})
