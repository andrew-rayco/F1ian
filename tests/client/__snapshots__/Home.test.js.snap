// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="twelve columns home"
      >
        <h3
          data-test="heading-seasons"
        >
          Seasons
        </h3>
        <ul
          className="seasons"
        >
          <Loading />
        </ul>
        <h3
          data-test="heading-circuits"
        >
          Circuits
        </h3>
        <div
          className="circuits"
        >
          <a
            href="/#/circuits"
          >
            See all the circuits
          </a>
        </div>
      </div>,
      "className": "row",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            data-test="heading-seasons"
          >
            Seasons
          </h3>,
          <ul
            className="seasons"
          >
            <Loading />
          </ul>,
          <h3
            data-test="heading-circuits"
          >
            Circuits
          </h3>,
          <div
            className="circuits"
          >
            <a
              href="/#/circuits"
            >
              See all the circuits
            </a>
          </div>,
        ],
        "className": "twelve columns home",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Seasons",
            "data-test": "heading-seasons",
          },
          "ref": null,
          "rendered": "Seasons",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Loading />,
            "className": "seasons",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Circuits",
            "data-test": "heading-circuits",
          },
          "ref": null,
          "rendered": "Circuits",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="/#/circuits"
            >
              See all the circuits
            </a>,
            "className": "circuits",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "See all the circuits",
              "href": "/#/circuits",
            },
            "ref": null,
            "rendered": "See all the circuits",
            "type": "a",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="twelve columns home"
        >
          <h3
            data-test="heading-seasons"
          >
            Seasons
          </h3>
          <ul
            className="seasons"
          >
            <Loading />
          </ul>
          <h3
            data-test="heading-circuits"
          >
            Circuits
          </h3>
          <div
            className="circuits"
          >
            <a
              href="/#/circuits"
            >
              See all the circuits
            </a>
          </div>
        </div>,
        "className": "row",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3
              data-test="heading-seasons"
            >
              Seasons
            </h3>,
            <ul
              className="seasons"
            >
              <Loading />
            </ul>,
            <h3
              data-test="heading-circuits"
            >
              Circuits
            </h3>,
            <div
              className="circuits"
            >
              <a
                href="/#/circuits"
              >
                See all the circuits
              </a>
            </div>,
          ],
          "className": "twelve columns home",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Seasons",
              "data-test": "heading-seasons",
            },
            "ref": null,
            "rendered": "Seasons",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Loading />,
              "className": "seasons",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "ul",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Circuits",
              "data-test": "heading-circuits",
            },
            "ref": null,
            "rendered": "Circuits",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                href="/#/circuits"
              >
                See all the circuits
              </a>,
              "className": "circuits",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "See all the circuits",
                "href": "/#/circuits",
              },
              "ref": null,
              "rendered": "See all the circuits",
              "type": "a",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
          "getChildContext": Object {
            "calledByRenderer": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
